### DISCUSSION TITLE
Emitting from an External Process -- Approaches

### DISCUSSION THREAD

**Discussion by Naman-ntc** [miguelgrinberg/Flask-SocketIO#1595] [Q&A]
According to documentation when emitting from external processes one can instantiate a `SocketIO` instance (given one is using a message queue) and emit using that. 
```
socketio = SocketIO(message_queue='redis://')
socketio.emit('my event', {'data': 'foo'}, namespace='/test')
```

However, another way is to cause an emit from the external process might be to fake a POST request to the server and and emit from that event handler. 

```
def extenal_process():
        post('SERVER/finished', json=meta)

@app.route('/finished/', methods=['POST'])
def event():
    userid,data = request.get_json()
    socketio.emit('celerystatus', data, namespace='/events', room=userid)
    return 'ok'
```

I was wondering if there is any downside to using the second approach or if you recommend both approaches? 

While I do prefer the former approach and realize that faking POST requests might hog the server, having a separate message queue and instantiating a new object every time might also hog the celery worker (my external process). 

Thanks

**Comment by miguelgrinberg** [comment]
Nothing wrong with using an HTTP request, but I don't understand why you say that you'll be instantiating a new object when you use the message queue method. The `socketio` object would be global, you wouldn't be creating a new one every time.

**Comment by Naman-ntc** [comment]
Oh, I think I misunderstood. My celery tasks were in a separate file and I was instantiating socketio inside my celery task for some reason. I will make socketio a global variable in my tasks.py.
